node/lib/util/stitch_util.js

```diff
@@ -778,8 +778,15 @@ exports.readConvertedCommits = co.wrap(function *(repo) {
 
     const result =
                  yield BulkNotesUtil.readNotes(repo, exports.convertedNoteRef);
-    for (let key in result) {
-        result[key] = exports.readConvertedContent(result[key]);
+    for (let key of Object.keys(result)) {
+        const sha = exports.readConvertedContent(result[key]);
+        try {
+            yield repo.getCommit(sha);
+        } catch (e) {
+            // We have the note but not the commit, delete from cache
+            delete result[key];
+        }
+        result[key] = sha;
```

Comment:
`result[key] = sha;` will always be executed, right? Do you want to move this inside try block?

Reply:
oops